### ISSUE TITLE
Color: `Darken()` produces unexpected result

### ISSUE THREAD

**Issue by swharden** [ScottPlot/ScottPlot#3387]
![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/30b85e8a-b8c3-4637-b9bb-316ab118ab9d)

border is expected to be darker than fill but it's a different hue

**Comment by KroMignon** [comment]
@swharden I am not totally sure the used code to calculate Hue, Saturation and Luminosity is consistent. I my understanding of this value they should be limited between:

* Hue: [0° .. 360°]
* Saturation: [0.0 .. 1.0]
* Luminosity: [0.0 .. 1.0]

That don't seems to be the case with current code:
```csharp

var c = Colors.Red;
var rgb = c.ToStringRGB();
(float h, float s, float l) = c1.FillStyle.Color.ToHSL();
Debug.WriteLine($"RGB: {rgb} Hue: {h}, Saturation: {s}, Liminance: {l}");
```

Result is: `RGB: #FF0000 Hue: 1, Saturation: -1,0079051, Liminance: 127,5`

**Comment by swharden** [comment]
Maybe this has been fixed with #3390 ... the following looks good to me!

```cs
using ScottPlot;

namespace Sandbox.WinForms;

public partial class Form1 : Form
{
    public Form1()
    {
        InitializeComponent();

        for (int i = 0; i < 10; i++)
        {
            double fraction = (double)i / 10;
            var circle = formsPlot1.Plot.Add.Circle(i, 0, 1);
            circle.LineWidth = 0;
            circle.FillColor = Colors.LightPink.Darken(fraction);
        }

        for (int i = 0; i < 10; i++)
        {
            double fraction = (double)i / 10;
            var circle = formsPlot1.Plot.Add.Circle(i, -2, 1);
            circle.LineWidth = 0;
            circle.FillColor = Colors.Navy.Lighten(fraction);
        }
    }
}
```

![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/a3848fd1-be8d-410a-8ded-01358d4fd0d3)
